calendar/api/post_action_test.go

```diff
@@ -0,0 +1,619 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/mattermost/mattermost-plugin-mscalendar/calendar/config"
+	"github.com/mattermost/mattermost-plugin-mscalendar/calendar/engine"
+	"github.com/mattermost/mattermost-plugin-mscalendar/calendar/remote"
+	"github.com/mattermost/mattermost-plugin-mscalendar/calendar/store"
+
+	"github.com/mattermost/mattermost/server/public/model"
+)
+
+func TestPreprocessAction(t *testing.T) {
+	api, _, _, _, _, _, _, _ := GetMockSetup(t)
+
+	tests := []struct {
+		name       string
+		setup      func(*http.Request)
+		assertions func(*httptest.ResponseRecorder, engine.Engine, *engine.User, string, string, string)
+	}{
+		{
+			name:  "Missing Mattermost user ID",
+			setup: func(req *http.Request) {},
+			assertions: func(rec *httptest.ResponseRecorder, mscal engine.Engine, user *engine.User, eventID, option, postID string) {
+				assert.Equal(t, http.StatusOK, rec.Result().StatusCode)
+				assert.Nil(t, mscal)
+				assert.Nil(t, user)
+				assert.Empty(t, eventID)
+				assert.Empty(t, option)
+				assert.Empty(t, postID)
+			},
+		},
+		{
+			name: "Invalid request body",
+			setup: func(req *http.Request) {
+				req.Header.Set(MMUserIDHeader, MockUserID)
+				req.Body = io.NopCloser(bytes.NewBufferString("invalid json"))
+			},
+			assertions: func(rec *httptest.ResponseRecorder, mscal engine.Engine, user *engine.User, eventID, option, postID string) {
+				assert.Equal(t, http.StatusOK, rec.Result().StatusCode)
```

Comment:
Why are we getting status OK in all the cases. Please check if we need to update our code to return proper status codes

Reply:
We are not writing any status code in the whole file. I think we should make it a separate task to return proper status code for all the API in the plugin

Reply:
@Kshitij-Katiyar Sure, please log a issue for the same